src/kernel/scheduler.zig

```diff
@@ -352,37 +354,55 @@ fn rt_variable_preserved(allocator: *Allocator) void {
 ///     IN mem_profile: mem.MemProfile - The system's memory profile. Determines the end address of the user task's VMM.
 ///
 fn rt_user_task(allocator: *Allocator, mem_profile: *const mem.MemProfile) void {
-    // 1. Create user VMM
-    var task_vmm = allocator.create(vmm.VirtualMemoryManager(arch.VmmPayload)) catch |e| {
-        panic(@errorReturnTrace(), "Failed to allocate user task VMM: {}\n", .{e});
-    };
-    task_vmm.* = vmm.VirtualMemoryManager(arch.VmmPayload).init(0, @ptrToInt(mem_profile.vaddr_start), allocator, arch.VMM_MAPPER, undefined) catch unreachable;
-    // 2. Create user task. The code will be loaded at address 0
-    var user_task = task.Task.create(0, false, task_vmm, allocator) catch |e| {
-        panic(@errorReturnTrace(), "Failed to create user task: {}\n", .{e});
-    };
-    // 3. Read the user program file from the filesystem
-    const user_program_file = fs.openFile("/user_program", .NO_CREATION) catch |e| {
-        panic(@errorReturnTrace(), "Failed to open /user_program: {}\n", .{e});
-    };
-    defer user_program_file.close();
-    var code: [1024]u8 = undefined;
-    const code_len = user_program_file.read(code[0..1024]) catch |e| {
-        panic(@errorReturnTrace(), "Failed to read user program file: {}\n", .{e});
-    };
-    // 4. Allocate space in the vmm for the user_program
-    const code_start = task_vmm.alloc(std.mem.alignForward(code_len, vmm.BLOCK_SIZE) / vmm.BLOCK_SIZE, .{ .kernel = false, .writable = true, .cachable = true }) catch |e| {
-        panic(@errorReturnTrace(), "Failed to allocate VMM memory for user program code: {}\n", .{e});
-    } orelse panic(null, "User task VMM didn't allocate space for the user program\n", .{});
-    if (code_start != 0) panic(null, "User program start address was {} instead of 0\n", .{code_start});
-    // 5. Copy user_program code over
-    vmm.kernel_vmm.copyData(task_vmm, code[0..code_len], code_start, true) catch |e| {
-        panic(@errorReturnTrace(), "Failed to copy user code: {}\n", .{e});
-    };
-    // 6. Schedule it
-    scheduleTask(user_task, allocator) catch |e| {
-        panic(@errorReturnTrace(), "Failed to schedule the user task: {}\n", .{e});
-    };
+    for (&[_][]const u8{ "user_program_data.elf", "user_program.elf" }) |user_program| {
```

Comment:
Why not include the `/` in the paths so you don't need the `std.mem.join`?

Reply:
Sure ting